fairworkflows/fairstep.py

```diff
@@ -409,22 +420,25 @@ def _modify_function(func):
     return _modify_function
 
 
-def _extract_inputs_from_function(func) -> List[FairVariable]:
+def _extract_inputs_from_function(func, additional_params) -> List[FairVariable]:
     """
     Extract inputs from function using inspection. The name of the argument will be the name of
     the fair variable, the corresponding type hint will be the type of the variable.
     """
     argspec = inspect.getfullargspec(func)
     try:
-        return [FairVariable(name=arg, type=argspec.annotations[arg].__name__)
-                for arg in argspec.args]
+        return [FairVariable(
+                    name=arg,
+                    computational_type=argspec.annotations[arg].__name__,
+                    types=additional_params.get(arg))
```

Comment:
This is quite a clean solution actually!

Reply:
:)